src/hooks/spoofing/write_player_gamer_data_node.cpp

```diff
@@ -3,36 +3,37 @@
 
 namespace big
 {
-	void hooks::write_player_gamer_data_node(rage::netObject* player, CPlayerGamerDataNode* node)
-	{
-		g_hooking->get_original<write_player_gamer_data_node>()(player, node);
+    void hooks::write_player_gamer_data_node(rage::netObject* player, CPlayerGamerDataNode* node)
+    {
+        g_hooking->get_original<write_player_gamer_data_node>()(player, node);
 
-		if (g.spoofing.spoof_crew_data)
-		{
-			node->m_clan_data.m_clan_member_id = 1;
-			node->m_clan_data.m_clan_id = 41564112;
-			node->m_clan_data.m_clan_id_2 = 41564112;
-			node->m_clan_data.m_clan_color = 420;
-			node->m_clan_data.m_clan_member_count = 15;
-			node->m_clan_data.m_clan_created_time = 420;
+        // move these variables to a separate function
+        const bool spoof_rockstar_id = g.spoofing.spoof_rockstar_id;
+        const bool spoof_rockstar_dev = spoof_rockstar_id && g.spoofing.spoof_rockstar_dev;
+        const bool spoof_rockstar_qa = spoof_rockstar_id && g.spoofing.spoof_rockstar_qa;
 
-			strcpy(node->m_clan_data.m_clan_tag, g.spoofing.crew_tag.c_str());
-			node->m_clan_data.m_is_system_clan = g.spoofing.rockstar_crew;
-			node->m_clan_data.m_is_clan_open = g.spoofing.square_crew_tag;
+        if (g.spoofing.spoof_crew_data)
+        {
+            node->m_clan_data.m_clan_member_id = 1;
+            node->m_clan_data.m_clan_id = 41564112;
+            node->m_clan_data.m_clan_id_2 = 41564112;
+            node->m_clan_data.m_clan_color = 420;
+            node->m_clan_data.m_clan_member_count = 15;
+            node->m_clan_data.m_clan_created_time = 420;
 
-			if (g.spoofing.rockstar_crew)
-			{
-				strcpy(node->m_clan_data.m_clan_name, "Rockstar");
-			}
-		}
+            strcpy(node->m_clan_data.m_clan_tag, g.spoofing.crew_tag.c_str());
```

Comment:
strcpy should be safe to use as std::string::c_str() is guaranteed to have string termination char, alternatively use memcpy that limits to the size of `m_clan_tag` if you're worried about writing too much.

Reply:
Decided to replace with memcpy.